test/integration/acceptance/create_domain_test.cpp

```diff
@@ -0,0 +1,219 @@
+  }
+
+  const std::string kRole = "role"s;
+  const std::string kUser = "user"s;
+  const std::string kNewDomain = "newdomain";
+  const std::string kUserId = kUser + "@test";
+  const crypto::Keypair kAdminKeypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+  const crypto::Keypair kUserKeypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command
+ * @then there is the tx in proposal
+ */
+TEST_F(CreateDomain, Basic) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(kNewDomain, kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * @given some user without can_create_domain permission
+ * @when execute tx with CreateDomain command
+ * @then there is no tx in proposal
+ */
+TEST_F(CreateDomain, NoPermissions) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms({iroha::model::can_get_my_txs}))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(kNewDomain, kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with inexistent domain
+ * @then there is no tx in proposal
+ */
+TEST_F(CreateDomain, NoRole) {
+  const std::string inexistent_role = "asdf"s;
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(kNewDomain, inexistent_role)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with already existing domain
+ * @then there is no tx in proposal
+ */
+TEST_F(CreateDomain, ExistentName) {
+  const std::string &existing_domain = IntegrationTestFramework::kDefaultDomain;
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(completeTx(baseTx().createDomain(existing_domain, kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with maximum available length
+ * @then there is no tx in proposal
```

Comment:
there is tx in proposal

Reply:
Please fix `@then` part of docstring and corresponding column for C243.1 in [Confluence](https://soramitsu.atlassian.net/wiki/spaces/IS/pages/299958273/Acceptance+Tests).